File: frontend/src/types/donation.ts
import type { Pagination } from './person';

export interface Donation {
  id: string;
  donor_person_id: string | null;
  campaign_id: string | null;
  campus_id: string;
  donation_type: string;
  amount: number | null;
  currency: string;
  item_description: string | null;
  donation_date: string;
  receipt_number: string | null;
  receipt_issued_at: string | null;
  notes: string | null;
  created_at: string;
  updated_at: string;
}

export interface DonationDetail extends Donation {
  donor_name?: string;
  campaign_name?: string;
}

export interface DonationListItem {
  id: string;
  donation_type: string;
  amount: number | null;
  currency: string;
  donation_date: string;
  donor_name?: string | null;
  campaign_name?: string | null;
}

export interface DonationListResponse {
  data: DonationListItem[];
  pagination: Pagination;
}

export interface DonationInput {
  donation_type: string;
  amount?: number;
  currency?: string;
  item_description?: string;
  donor_person_id?: string;
  campaign_id?: string;
  donation_date?: string;
  notes?: string;
}

export type UpdateDonationInput = DonationInput;

File: backend/internal/domain/donation.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

// Donation represents a contribution (financial, goods, or services) recorded
// against a campus, optionally linked to a donor and a campaign.
type Donation struct {
	ID              uuid.UUID  `json:"id"`
	DonorPersonID   *uuid.UUID `json:"donor_person_id,omitempty"`
	CampaignID      *uuid.UUID `json:"campaign_id,omitempty"`
	CampusID        uuid.UUID  `json:"campus_id"`
	DonationType    string     `json:"donation_type"`
	Amount          *float64   `json:"amount,omitempty"`
	Currency        string     `json:"currency"`
	ItemDescription *string    `json:"item_description,omitempty"`
	DonationDate    time.Time  `json:"donation_date"`
	ReceiptNumber   *string    `json:"receipt_number,omitempty"`
	ReceiptIssuedAt *time.Time `json:"receipt_issued_at,omitempty"`
	Notes           *string    `json:"notes,omitempty"`
	RegisteredBy    uuid.UUID  `json:"registered_by"`
	CreatedAt       time.Time  `json:"created_at"`
	UpdatedAt       time.Time  `json:"updated_at"`
}

// DonationDetail is the full donation representation with resolved names for
// the linked donor and campaign.
type DonationDetail struct {
	Donation
	DonorName    *string `json:"donor_name,omitempty"`
	CampaignName *string `json:"campaign_name,omitempty"`
}

// DonationListItem is a summary representation for list responses.
type DonationListItem struct {
	ID           uuid.UUID `json:"id"`
	DonationType string    `json:"donation_type"`
	Amount       *float64  `json:"amount,omitempty"`
	Currency     string    `json:"currency"`
	DonationDate time.Time `json:"donation_date"`
	DonorName    *string   `json:"donor_name,omitempty"`
	CampaignName *string   `json:"campaign_name,omitempty"`
}

// DonationListResult contains paginated donation list data.
type DonationListResult struct {
	Data       []DonationListItem `json:"data"`
	Pagination Pagination         `json:"pagination"`
}

// DonationFilter defines search/filter criteria for listing donations.
type DonationFilter struct {
	CampusID     uuid.UUID
	DonationType *string
	CampaignID   *uuid.UUID
	Page         int
	PerPage      int
}

// DonationTypes enumerates the valid donation categories.
var DonationTypes = []string{"FINANCIAL", "GOODS", "SERVICES"}

// DonationCurrencies enumerates the supported ISO 4217 currency codes.
var DonationCurrencies = []string{"BRL", "USD", "EUR"}
